server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -370,13 +402,19 @@ public DesiredBalance compute(
 
             logger.log(
                 reportByIterationCount || reportByTime ? Level.INFO : i % 100 == 0 ? Level.DEBUG : Level.TRACE,
-                () -> Strings.format(
-                    "Desired balance computation for [%d] is still not converged after [%s] and [%d] iterations",
-                    desiredBalanceInput.index(),
-                    TimeValue.timeValueMillis(currentTime - computationStartedTime).toString(),
-                    iterations
-                )
+                "Desired balance computation for [{}] is still not converged after [{}] and [{}] iterations, "
```

Comment:
I think I'd rather keep this one lazy, i.e. using the `() -> Strings.format(...` that was there before. No need to allocate two `TimeValue` instances plus their string representations plus the arguments array if we're just going to throw them all away because `DEBUG` logging is disabled. Not that I expect it really matters that much here but it's a good habit to acquire.